libsass uses (deprecated) sprintf

Discovered via BDR's M1mac checks https://www.stats.ox.ac.uk/pub/bdr/M1mac/sass.out

```
* checking whether package ‘sass’ can be installed ... [54s/58s] WARNING
Found the following significant warnings:
  src/json.cpp:1289:3: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
See ‘/Users/ripley/R/packages/tests-devel/sass.Rcheck/00install.out’ for details.
```

CRAN has requested a fix within the next 2 weeks

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -39,7 +39,7 @@ License: MIT + file LICENSE
 URL: https://rstudio.github.io/sass/, https://github.com/rstudio/sass
 BugReports: https://github.com/rstudio/sass/issues
 Encoding: UTF-8
-RoxygenNote: 7.2.0
+RoxygenNote: 7.2.1
 Roxygen: list(markdown = TRUE)
 SystemRequirements: GNU make
 Imports:

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -39,7 +39,7 @@ License: MIT + file LICENSE
 URL: https://rstudio.github.io/sass/, https://github.com/rstudio/sass
 BugReports: https://github.com/rstudio/sass/issues
 Encoding: UTF-8
-RoxygenNote: 7.2.1
+RoxygenNote: 7.2.2
 Roxygen: list(markdown = TRUE)
 SystemRequirements: GNU make
 Imports:

--- a/man/sass-deprecated.Rd
+++ b/man/sass-deprecated.Rd
@@ -12,7 +12,7 @@ Deprecated Sass functions
 }
 \section{Functions}{
 \itemize{
-\item \code{sass_layer_merge}: Please use \code{sass_bundle(...)}
-}}
+\item \code{sass_layer_merge()}: Please use \code{sass_bundle(...)}
 
+}}
 \keyword{internal}

--- a/man/sass_layer.Rd
+++ b/man/sass_layer.Rd
@@ -85,17 +85,17 @@ earlier layers, effectively 'new' defaults through all the 'old' defaults.
 }
 \section{Functions}{
 \itemize{
-\item \code{sass_layer}: Compose the parts of a single Sass layer. Object returned is a \code{sass_bundle()} with a single Sass layer
+\item \code{sass_layer()}: Compose the parts of a single Sass layer. Object returned is a \code{sass_bundle()} with a single Sass layer
 
-\item \code{sass_layer_file}: Read in a \code{.scss} file with parse special \verb{/*-- scss:(functions|defaults|rules|mixins) --*/} comments as relevant sections of a \code{sass_layer()}.
+\item \code{sass_layer_file()}: Read in a \code{.scss} file with parse special \verb{/*-- scss:(functions|defaults|rules|mixins) --*/} comments as relevant sections of a \code{sass_layer()}.
 
-\item \code{sass_bundle}: Collect \code{sass_bundle()} and/or \code{sass_layer()} objects. Unnamed Sass bundles will be concatenated together, preserving their internal name structures. Named Sass bundles will be condensed into a single Sass layer for easier removal from the returned Sass bundle.
+\item \code{sass_bundle()}: Collect \code{sass_bundle()} and/or \code{sass_layer()} objects. Unnamed Sass bundles will be concatenated together, preserving their internal name structures. Named Sass bundles will be condensed into a single Sass layer for easier removal from the returned Sass bundle.
 
-\item \code{sass_bundle_remove}: Remove a whole \code{sass_layer()} from a \code{sass_bundle()} object.
+\item \code{sass_bundle_remove()}: Remove a whole \code{sass_layer()} from a \code{sass_bundle()} object.
 
-\item \code{is_sass_bundle}: Check if \code{x} is a Sass bundle object
+\item \code{is_sass_bundle()}: Check if \code{x} is a Sass bundle object
+
 }}
-
 \examples{
 blue <- list(color = "blue !default")
 red <- list(color = "red !default")

--- a/scripts/patches/009-deprecated-declarations.patch
+++ b/scripts/patches/009-deprecated-declarations.patch
@@ -1,0 +1,13 @@
+diff --git a/src/libsass/src/json.cpp b/src/libsass/src/json.cpp
+index f7d06e4..41ee2f1 100644
+--- a/src/libsass/src/json.cpp
++++ b/src/libsass/src/json.cpp
+@@ -1286,7 +1286,7 @@ static void emit_number(SB *out, double num)
+    * like 0.3 -> 0.299999999999999988898 .
+    */
+   char buf[64];
+-  sprintf(buf, "%.16g", num);
++  snprintf(buf, sizeof(buf), "%.16g", num);
+ 
+   if (number_is_valid(buf))
+     sb_puts(out, buf);

--- a/src/libsass/src/json.cpp
+++ b/src/libsass/src/json.cpp
@@ -1286,7 +1286,7 @@ static void emit_number(SB *out, double num)
    * like 0.3 -> 0.299999999999999988898 .
    */
   char buf[64];
-  sprintf(buf, "%.16g", num);
+  snprintf(buf, sizeof(buf), "%.16g", num);
 
   if (number_is_valid(buf))
     sb_puts(out, buf);

